File: js/stage3/S3_OBJ_05.js
// S3_OBJ_05 — Invert
function invert(obj) {
  const result = {};
  for (const key in obj) {
    const value = obj[key];
    if (result[value]) {
      result[value].push(key);
    } else {
      result[value] = [key];
    }
  }
  return result;
}
// Test
console.log(invert({ a: 1, b: 2, c: 1 }));

File: Python/stage3/S3_OBJ_05.py
# S3_OBJ_05 — Invert dictionary
def invert(obj):
   result = {}
   for k, v in obj.items():
       if v in result:
           if isinstance(result[v], list):
               result[v].append(k)
           else:
               result[v] = [result[v], k]
       else:
           result[v] = k
   return result
print(invert({"a": 1, "b": 2, "c": 1}))